src/components/Account/Templates/Templates.vue

```diff
@@ -82,17 +99,23 @@ export default {
             this.loading = false;
 
             this.$store.commit("loading", false);
-            this.$store.commit('title', this.title);
         },
 
         refresh () {
+            this.templates = [];
             this.loading = true;
+            SessionCache.invalidateTemplates();
             this.fetchTemplates();
         },
 
         createTemplate () {
             this.$router.push({ name: 'create-template' });
+        },
+
+        selectTemplate (id) {
+            this.selectedTemplate = this.templates.find(template => template.device_id == id);
```

Comment:
```suggestion
            this.selectedTemplate = this.templates.find(template => template.device_id == id);
            this.$emit('selected-template-text', this.selectedTemplate.text);
```
See above.

Additionally - why use `find` here? Index the templates according to `device_id` and save the loop. You can do this in `processTemplate`